homeassistant/components/thread/__init__.py

```diff
@@ -6,14 +6,20 @@
 from homeassistant.helpers.typing import ConfigType
 
 from .const import DOMAIN
-from .dataset_store import DatasetEntry, async_add_dataset, async_get_preferred_dataset
+from .dataset_store import (
+    DatasetEntry,
+    async_add_dataset,
+    async_get_preferred_dataset,
+    async_set_preferred_dataset,
+)
 from .websocket_api import async_setup as async_setup_ws_api
 
 __all__ = [
     "DOMAIN",
     "DatasetEntry",
     "async_add_dataset",
     "async_get_preferred_dataset",
+    "async_set_preferred_dataset",
```

Comment:
we should not expose this to other integrations. That's up to the interaction between user and panel.